org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -201,13 +202,23 @@ public void actionZoom()
 	{
 		String s = tableIDValue != null ? String.valueOf(tableIDValue) : "";
 		int tableID = s.length() > 0 ? Integer.parseInt(s) : 0;
-		s = recordIDValue != null ? String.valueOf(recordIDValue) : "";
-		int recordID = s.length() > 0 ? Integer.parseInt(s) : 0;
-		if(tableID <= 0 || recordID < 0)
-			return;
-		if (!MRole.getDefault().isTableAccess(tableID, false))
-			throw new AdempiereException(Msg.getMsg(Env.getCtx(), "AccessTableNoView"));
-		AEnv.zoom(tableID, recordID);
+		MTable table = MTable.get(tableID);
+		if (table.isUUIDKeyTable()) {
+			String recordUU = recordIDValue != null ? recordIDValue.toString() : "";
+			if(tableID <= 0)
```

Comment:
perhaps needs to check length of recordUU here ?

Reply:
> perhaps needs to check length of recordUU here ?

When there is no key the zoom function is opening the window without a record selected.  I didn't want to change this behavior.

Reply:
ok but that make it inconsistent with the else block below.

Reply:
it's OK - the else block also zoom to the window when the recordID == 0